packages/trpc/server/routers/viewer/slots/util.ts

```diff
@@ -89,63 +93,130 @@ export async function getEventType(input: TGetScheduleInputSchema) {
       }
     }
 
-    const eventType = await prisma.eventType.findFirst({
+    const startFirstPrismaEventTypeGet = performance.now();
+    eventType = await prisma.eventType.findFirst({
```

Comment:
I'd start off by preferring the eventTypeId - also if the eventTypeId exists you can probably safely to an findUniqueOrThrow to bail early, then only if there is no event type yet in the follow up try by slug+userId/teamId.

Reply:
So the problem is this code gets reached first because the eventTypeId doesn't exist so you have to find it searching by slug, teamId and userId. Are you thinking of a different workaround?